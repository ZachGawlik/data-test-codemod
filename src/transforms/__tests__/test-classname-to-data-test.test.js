const { transformFixture } = require('../../testUtils');

// Prevent escaping double-quotes, e.g. className=\\"btn\\"
const rawSerializer = {
  print: val => val,
  test: val => !!val
};
expect.addSnapshotSerializer(rawSerializer);

describe('test-classname-to-data-test', () => {
  it('defaults to transform test-* classnames to data-test attrbutes', () => {
    expect(
      transformFixture(__dirname, {}, 'test-classname-to-data-test.js')
    ).toMatchSnapshot();
  });
});
