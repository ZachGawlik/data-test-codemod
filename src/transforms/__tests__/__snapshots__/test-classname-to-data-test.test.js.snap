// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-classname-to-data-test defaults to transform test-* classnames to data-test attrbutes 1`] = `
import React from 'react';

const ButtonGroup = ({ editItem }) => (
  <div>
    <button data-test="add-btn">Add</button>
    <button className="btn btn-secondary" onClick={editItem} data-test="edit-btn">
      Edit
    </button>
    <button className="btn btn-danger" data-test="delete-btn">
      Delete
    </button>
  </div>
);

export default ButtonGroup;

`;
